lib/core/CJsonLogLayout.cc

```diff
@@ -63,132 +59,103 @@ bool CJsonLogLayout::locationInfo() const {
     return m_LocationInfo;
 }
 
-void CJsonLogLayout::properties(bool properties) {
-    m_Properties = properties;
-}
-
-bool CJsonLogLayout::properties() const {
-    return m_Properties;
-}
-
-void CJsonLogLayout::activateOptions(Pool& /*p*/) {
-    // NO-OP
-}
-
-void CJsonLogLayout::setOption(const LogString& option, const LogString& value) {
-    if (StringHelper::equalsIgnoreCase(option, LOG4CXX_STR("LOCATIONINFO"),
-                                       LOG4CXX_STR("locationinfo"))) {
-        this->locationInfo(OptionConverter::toBoolean(value, false));
-    }
-    if (StringHelper::equalsIgnoreCase(option, LOG4CXX_STR("PROPERTIES"),
-                                       LOG4CXX_STR("properties"))) {
-        this->properties(OptionConverter::toBoolean(value, false));
-    }
-}
-
-void CJsonLogLayout::format(LogString& output, const spi::LoggingEventPtr& event, Pool& /*p*/) const {
+void CJsonLogLayout::operator()(const boost::log::record_view& rec,
+                                boost::log::formatting_ostream& strm) const {
     using TStringBufferWriter = rapidjson::Writer<rapidjson::StringBuffer>;
     rapidjson::StringBuffer buffer;
-    TStringBufferWriter writer(buffer);
+    TStringBufferWriter writer{buffer};
 
     writer.StartObject();
 
     writer.String(LOGGER_NAME);
-    LOG4CXX_ENCODE_CHAR(logger, event->getLoggerName());
-    writer.String(logger);
+    writer.String(LOGGER);
 
     writer.String(TIMESTAMP_NAME);
-    writer.Int64(event->getTimeStamp() / 1000);
+    const boost::posix_time::ptime& timeStamp{
```

Comment:
nit: the uses the uses of extract in this file it seems like a good candidates for auto since the extract already defines the values' types
```suggestion
    const auto& timeStamp =
```